lib/ipfix_packet_elements/template_set/field_specifier.js

```diff
@@ -1,10 +1,18 @@
+const PEN_FLAG_VALUE = 32768;
 var TemplateRecordFieldSpecifier = function(buffer) {
     var self = this;
     this.fieldId = undefined;
     this.fieldLength = undefined;
+    this.enterpriseId = undefined;
+    this.enterprisefieldId = undefined;
     var _construct = function() {
         self.fieldId = buffer.readUInt16BE(0);
         self.fieldLength = buffer.readUInt16BE(2);
+        if (self.fieldId >= PEN_FLAG_VALUE){ // The PEN flag is on
```

Comment:
If it includes the flag, that means there's an extra 4 bytes of EnterpriseID following the length field
Enterprise fields can "override" the normal field ID, which is why we're handling it differently and assigning an "enterprise" field id in the normal field id (to avoid confusion when parsing)